source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs

```diff
@@ -35,15 +39,37 @@ public record JournalEntry
 
             EnsureDebitsAndCreditsBalance(debits, credits);
 
+            UserId = userId;
             Date = date;
             Debits = new List<JournalEntryDebitLine>(debits);
             Credits = new List<JournalEntryCreditLine>(credits);
         }
 
+        public string UserId { get; }
         public DateTime Date { get; }
         public string Description { get; }
         public List<JournalEntryDebitLine> Debits { get; }
         public List<JournalEntryCreditLine> Credits { get; }
+        public bool Closed { get; private set; }
+
+        public ImmutableList<Review> Reviews { get; private set; }
+
+        public void AddReview(Review review)
+        {
+            if (Closed)
+                throw new InvalidOperationException("Journal is already approved and closed");
+
+            if (string.Compare(UserId, review.Reviewer, true) == 0)
+                throw new ArgumentException("Accountant cannot review and approve own journal entry", nameof(review.Reviewer));
+
+            if (review.Approved)
+                Closed = true; //maybe emit event so a service can save journal
```

Comment:
What component would benefit from this event?

Reply:
It is just a suggestion. This is about a pattern I read sometime ago, but we do not have to implement it if you don't see it befitting. The idea is to make services subscribe to an event that is triggered whenever a journal entry is approved. it could be a service that persists a journal entry to the db, and another service that sends an email or a notification on the screen (perhaps through SignalR) of a user, instead of calling or instantiating those services. My understanding of events and design patterns is shallow, so I wanted to see what you thought about it.

Reply:
Ah I like this! I think it's a good idea to use events, but let's wait until we have someone who might consume them. But otherwise, events are pretty simple in dotnet thankfully. I appreciate the suggestion.